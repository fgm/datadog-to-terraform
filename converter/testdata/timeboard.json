{
  "title": "Laura's Timeboard 21 Aug 2020 16:30",
  "description": "",
  "widgets": [
    {
      "id": 7042236314922270,
      "definition": {
        "type": "timeseries",
        "requests": [
          {
            "q": "avg:system.cpu.user{*}",
            "display_type": "line",
            "style": { "palette": "dog_classic", "line_type": "solid", "line_width": "normal" }
          }
        ],
        "yaxis": {
          "label": "",
          "scale": "linear",
          "min": "auto",
          "max": "auto",
          "include_zero": true
        },
        "title": "Avg of system.cpu.user over *",
        "show_legend": false,
        "legend_size": "0"
      }
    },
    {
      "id": 4204614699449750,
      "definition": {
        "type": "query_value",
        "requests": [
          {
            "q": "avg:datadog.agent.running{*}",
            "aggregator": "avg",
            "conditional_formats": [
              { "comparator": "<", "value": 1, "palette": "white_on_red" },
              { "comparator": ">=", "value": 1, "palette": "white_on_green" }
            ]
          }
        ],
        "title": "Avg of datadog.agent.running over *",
        "time": {},
        "autoscale": true,
        "precision": 2
      }
    }
  ],
  "template_variables": [
    { "name": "major", "default": "*", "prefix": "agent_version_major" },
    { "name": "minor", "default": "*", "prefix": "agent_version_minor" },
    { "name": "patch", "default": "*", "prefix": "agent_version_patch" }
  ],
  "layout_type": "ordered",
  "is_read_only": false,
  "notify_list": [],
  "template_variable_presets": [
    { "name": ">= 7", "template_variables": [{ "name": "major", "value": "7" }] },
    {
      "name": "Latest",
      "template_variables": [
        { "name": "major", "value": "7" },
        { "name": "minor", "value": "21" },
        { "name": "patch", "value": "1" }
      ]
    }
  ],
  "id": "hww-tah-72j"
}
