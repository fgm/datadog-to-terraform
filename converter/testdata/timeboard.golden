resource "datadog_dashboard" "timeboard" {
description = ""
is_read_only = false
layout_type = "ordered"
notify_list = []

template_variable_preset {
name = ">= 7"

template_variable {
name = "major"
value = "7"
}
}
template_variable_preset {
name = "Latest"

template_variable {
name = "major"
value = "7"
}
template_variable {
name = "minor"
value = "21"
}
template_variable {
name = "patch"
value = "1"
}
}

template_variable {
default = "*"
name = "major"
prefix = "agent_version_major"
}
template_variable {
default = "*"
name = "minor"
prefix = "agent_version_minor"
}
template_variable {
default = "*"
name = "patch"
prefix = "agent_version_patch"
}
title = "Laura's Timeboard 21 Aug 2020 16:30"

widget {

timeseries_definition {
legend_size = "0"

request {
display_type = "line"
q = "avg:system.cpu.user{*}"

style {
line_type = "solid"
line_width = "normal"
palette = "dog_classic"
}
}
show_legend = false
title = "Avg of system.cpu.user over *"
yaxis {
include_zero = true
label = ""
max = "auto"
min = "auto"
scale = "linear"
}
}
}
widget {

query_value_definition {
autoscale = true
precision = 2

request {
aggregator = "avg"

conditional_formats {
comparator = "<"
palette = "white_on_red"
value = 1
}
conditional_formats {
comparator = ">="
palette = "white_on_green"
value = 1
}
q = "avg:datadog.agent.running{*}"
}
title = "Avg of datadog.agent.running over *"
}
}
}